datalad/tests/fixtures.py

```diff
@@ -0,0 +1,73 @@
+"""utilities to assist with setting up tests"""
+
+from datalad.api import clone
+from datalad.core.distributed.clone import decode_source_spec
+from datalad.distribution.dataset import Dataset
+from datalad.interface.common_cfg import dirs
+from datalad.utils import Path, better_wraps, optional_args
+from datalad.tests.utils import with_tempfile
+
+
+DATALAD_TESTS_CACHE = Path(dirs.user_cache_dir) / "tests"
+
+
+def get_cached_dataset(url, dataset_name=None, version=None):
+    """ Helper to get a cached clone from url
+
+    Intended for use from within `datalad_dataset` decorator.
+    Clones `url` into user's cache under datalad/tests/`name`. If such a clone
+    already exists, don't clone but return the existing one. So, it's supposed
+    to cache the original source in order to reduce time and traffic for tests,
+    by letting subsequent requests clone from a local location directly.
+
+    If it's an annex get the content, too.
+
+    Note
+    ----
+    `version` not yet implemented. Intention: verify that `version` can be
```

Comment:
Most likely then version should be included in the cache path, since different tests might want different versions... Alternative is to checkout a version for the test, but it would prohibit running in parallel ever

Reply:
See comment below about necessity of `@datalad_dataset` always delivering a clone. I'd prefer to checkout a particular version only in that clone, that is delivered to the test, while the cached dataset itself only replaces the original source.